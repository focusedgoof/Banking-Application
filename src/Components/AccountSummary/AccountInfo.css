.accountInfo > h3 {
  display: flex;
  font-size: 30px;
  width: 100%;
  font-weight: bolder;
  justify-content: center;
  margin-top:0;
  margin-bottom:3%;
  color:#34b3f7;
}