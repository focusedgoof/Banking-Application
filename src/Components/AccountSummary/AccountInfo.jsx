// accountinfo.jsx
import React, { useEffect, useState } from "react";
import { getLocalStorage } from '../../Utils/Utils.jsx';
import './AccountInfo.css';

const AccountInfo = ({ balance }) => {
    const [loggedUserName, setLoggedUserName] = useState('');
    const [userData, setUserData] = useState(null);
    const cardDetails = [
        { key: 'showAccountNo', label: 'Account Number', value: userData?.AccountNumber },
        { key: 'showIFSC', label: 'IFSC Code', value: userData?.IFSCCode },
        { key: 'showBranch', label: 'Branch Details', value: userData?.Branch },
        { key: 'showBalance', label: 'Available Balance', value: `Rs. ${balance || userData?.Balance}` },
        { key: 'showPAN', label: 'PAN Details', value: userData?.PAN },
    ];
    const [cardStates, setCardStates] = useState({
        showAccountNo: false,
        showIFSC: false,
        showBranch: false,
        showBalance: false,
        showPAN: false,
    });


    useEffect(() => {
        const fetchData = async () => {
            try {
                const userDataFromSession = await getLocalStorage('userData');
                if (userDataFromSession) {
                    setUserData(userDataFromSession);
                    setLoggedUserName(userDataFromSession.Username);
                }
            } catch (error) {
                console.error('Error fetching user details:', error);
            }
        };

        fetchData();
    }, []);

    const toggleCard = (cardKey) => {
        setCardStates(prevState => ({
            ...prevState,
            [cardKey]: !prevState[cardKey]
        }));
    };

    

    return (
        <div className="accountInfo">
            <h3>Welcome, {loggedUserName || 'User'}. Here are your account details</h3>
            <div className="info-cards">
                {cardDetails.map(card => (
                    <div className="card" key={card.key}>
                        <p>{card.label}</p>
                        {!cardStates[card.key] ? (
                            <button onClick={() => toggleCard(card.key)}>Click to view</button>
                        ) : (
                            <p className="view-detail">{card.value}</p>
                        )}
                    </div>
                ))}
            </div>
        </div>
    );
};

export default AccountInfo;
