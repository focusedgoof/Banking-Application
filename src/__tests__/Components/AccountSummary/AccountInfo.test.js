import React from 'react';
import { render } from '@testing-library/react';
import AccountInfo from '../../../Components/AccountSummary/AccountInfo';

describe('AccountInfo Component', () => {
  it('matches snapshot', () => {
    const mockUserData = {
      Username: 'JohnDoe',
      AccountNumber: '123456789',
      IFSCCode: 'ABC123',
      Branch: 'Main Branch',
      Balance: 1000,
      PAN: 'ABCDE1234F',
    };

    const { asFragment } = render(<AccountInfo balance={mockUserData.Balance} />);

    expect(asFragment()).toMatchSnapshot();
  });
});