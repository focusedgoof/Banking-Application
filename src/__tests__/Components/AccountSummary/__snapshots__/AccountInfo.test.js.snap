// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="accountInfo"
  >
    <h3>
      Welcome, User. Here are your account details
    </h3>
    <div
      class="info-cards"
    >
      <div
        class="card"
      >
        <p>
          Account Number
        </p>
        <button>
          Click to view
        </button>
      </div>
      <div
        class="card"
      >
        <p>
          IFSC Code
        </p>
        <button>
          Click to view
        </button>
      </div>
      <div
        class="card"
      >
        <p>
          Branch Details
        </p>
        <button>
          Click to view
        </button>
      </div>
      <div
        class="card"
      >
        <p>
          Available Balance
        </p>
        <button>
          Click to view
        </button>
      </div>
      <div
        class="card"
      >
        <p>
          PAN Details
        </p>
        <button>
          Click to view
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
